diff -ru bwa-0.7.15-original/bwt.h bwa-0.7.15/bwt.h
--- bwa-0.7.15-original/bwt.h	2022-05-10 23:25:57.272241487 +0900
+++ bwa-0.7.15/bwt.h	2022-05-10 23:26:10.624277748 +0900
@@ -42,6 +42,7 @@
 #endif
 
 typedef uint64_t bwtint_t;
+typedef uint32_t u_int32_t;
 
 typedef struct {
 	bwtint_t primary; // S^{-1}(0), or the primary index of BWT
diff -ru bwa-0.7.15-original/bwt_lite.c bwa-0.7.15/bwt_lite.c
--- bwa-0.7.15-original/bwt_lite.c	2022-05-10 23:25:57.272241487 +0900
+++ bwa-0.7.15/bwt_lite.c	2022-05-10 23:26:10.624277748 +0900
@@ -7,6 +7,8 @@
 #  include "malloc_wrap.h"
 #endif
 
+typedef uint32_t u_int32_t;
+
 int is_sa(const uint8_t *T, int *SA, int n);
 int is_bwt(uint8_t *T, int n);
 
diff -ru bwa-0.7.15-original/kthread.c bwa-0.7.15/kthread.c
--- bwa-0.7.15-original/kthread.c	2022-05-10 23:25:57.272241487 +0900
+++ bwa-0.7.15/kthread.c	2022-05-10 23:26:10.624277748 +0900
@@ -1,6 +1,7 @@
 #include <pthread.h>
 #include <stdlib.h>
 #include <limits.h>
+#include <stdint.h>
 
 /************
  * kt_for() *
diff -ru bwa-0.7.15-original/rle.h bwa-0.7.15/rle.h
--- bwa-0.7.15-original/rle.h	2022-05-10 23:25:57.272241487 +0900
+++ bwa-0.7.15/rle.h	2022-05-10 23:26:23.712313298 +0900
@@ -30,7 +30,7 @@
  *** 43+3 codec ***
  ******************/
 
-const uint8_t rle_auxtab[8];
+extern const uint8_t rle_auxtab[8];
 
 #define RLE_MIN_SPACE 18
 #define rle_nptr(block) ((uint16_t*)(block))
